src/google/protobuf/metadata.h

```diff
@@ -59,8 +59,7 @@ namespace internal {
 class LIBPROTOBUF_EXPORT InternalMetadataWithArena {
```

Comment:
Actually, I just did some quick testing locally and one of the targets that was failing with the upstream sources will build successfully if I just follow your suggestion of not marking this class as LIBPROTOBUF_EXPORT.  The new patch simply makes that change.

I found one other fix that seems to work as well: add a metadata.cc and #include metadata.h from it, but don't actually de-inline any methods into it.  I assume that for some reason the compiler decides metadata.obj is where it should put the out-of-line copies of these symbols for export.  We can use this method if it turns out the LIBPROTOBUF_EXPORT declaration is necessary after all.